evm_loader/shared-lib/src/lib.rs

```diff
@@ -0,0 +1,297 @@
+            let config = Arc::new(create_from_api_comnfig(&api_options)?);
+            let rpc_client = context::build_rpc_client(&config, slot)?;
+            let context = context::create(rpc_client, config.clone());
+            let rpc_client = context
+                .rpc_client
+                .as_any()
+                .downcast_ref::<RpcClient>()
+                .unwrap();
+            let signer = context.signer()?;
+            Ok(serde_json::to_string(
+                &deposit::execute(
+                    rpc_client,
+                    config.evm_loader,
+                    &signer,
+                    amount,
+                    &ether_address,
+                )
+                .await?,
+            )
+            .unwrap())
+        }
+        "emulate" => {
+            let param_types::Params {
+                api_options,
+                slot,
+                params:
+                    param_types::Emulate {
+                        tx_params,
+                        token_mint,
+                        chain_id,
+                        step_limit,
+                        commitment,
+                        accounts,
+                        solana_accounts,
+                        trace_call_config,
+                    },
+            } = serde_json::from_str(params).map_err(|_| params_to_neon_error(params))?;
+            let config = Arc::new(create_from_api_comnfig(&api_options)?);
+            let rpc_client = context::build_rpc_client(&config, slot)?;
+            Ok(serde_json::to_string(
+                &emulate::execute(
+                    rpc_client.as_ref(),
+                    config.evm_loader,
+                    tx_params,
+                    token_mint,
+                    chain_id,
+                    step_limit,
+                    commitment,
+                    &accounts,
+                    &solana_accounts,
+                    trace_call_config,
+                )
+                .await?,
+            )
+            .unwrap())
+        }
+        "get_ether_account_data" => {
+            let param_types::Params {
+                api_options,
+                slot,
+                params: param_types::GetEtherAccountData { ether_address },
+            } = serde_json::from_str(params).map_err(|_| params_to_neon_error(params))?;
+            let config = Arc::new(create_from_api_comnfig(&api_options)?);
+            let rpc_client = context::build_rpc_client(&config, slot)?;
+            Ok(serde_json::to_string(
+                &get_ether_account_data::execute(
+                    rpc_client.as_ref(),
+                    &config.evm_loader,
+                    &ether_address,
+                )
+                .await?,
+            )
+            .unwrap())
+        }
+        "get_neon_elf" => {
+            let param_types::Params {
+                api_options,
+                slot,
+                params: param_types::GetNeonElf { program_location },
```

Comment:
I think, it's not a good idea to give unauthorized user access to arbitrary file on remote server. Pass just `None` as `program_location` in order to get ELF from Neon account only.